src/v/compression/async_stream_zstd.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "bytes/iobuf.h"
+#include "utils/mutex.h"
+
+#include <seastar/core/aligned_buffer.hh>
+
+#include <memory>
+#include <zstd.h>
+
+namespace compression {
+class async_stream_zstd {
```

Comment:
This class should be commented.

In fact, you have a very nice comment in the cover letter which would form a reasonable basis for the class comment.

Reply:
Added a class comment